components/lib/utils/FindChildrenByKey.js

```diff
@@ -0,0 +1,15 @@
+export const findChildrenByKey = (data, key) => {
+    for (const item of data) {
+        if (item.key === key) {
+            return item.children || [];
+        } else if (item.children) {
+            const result = findChildrenByKey(item.children, key);
+
+            if (result.length > 0) {
+                return result;
+            }
+        }
+    }
+
+    return [];
```

Comment:
you can probably put this in ObjectUtils.js to reduce class sprawl...

Reply:
Updated